rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1251,20 +1251,54 @@ private void processReleaseTransactions(Transaction rskTx) {
-                // therefore it won't have a rskTxHash value, fallback to this transaction's hash
-                txsWaitingForSignatures.put(entry.getRskTxHash() == null ? rskTx.getHash() : entry.getRskTxHash(), entry.getTransaction());
-            }
-            else {
-                txsWaitingForSignatures.put(rskTx.getHash(), entry.getTransaction());
-            }
-            if(activations.isActive(ConsensusRule.RSKIP326)) {
-                eventLogger.logPegoutConfirmed(entry.getTransaction().getHash(), entry.getRskBlockNumber());
-            }
+        if (txsWithEnoughConfirmations.isEmpty()) {
+            return;
+        }
+
+        ReleaseTransactionSet.Entry entry = txsWithEnoughConfirmations.iterator().next();
+
+        Keccak256 txWaitingForSignatureKey = getTxWaitingForSignatureKey(rskTx, entry);
+        if (activations.isActive(ConsensusRule.RSKIP375)){
+            checkIfEntryExistsInTxsWaitingForSignatures(txWaitingForSignatureKey, entry.getTransaction(), txsWaitingForSignatures);
+        }
+
+        txsWaitingForSignatures.put(txWaitingForSignatureKey, entry.getTransaction());
+
+        if(activations.isActive(ConsensusRule.RSKIP326)) {
+            eventLogger.logPegoutConfirmed(entry.getTransaction().getHash(), entry.getRskBlockNumber());
+        }
+    }
+
+    private Keccak256 getTxWaitingForSignatureKey(Transaction rskTx, ReleaseTransactionSet.Entry entry) {
+        Keccak256 rskTxHash;
+        if (activations.isActive(ConsensusRule.RSKIP375)){
+            rskTxHash = entry.getRskTxHash();
+        }
+        // Since RSKIP176 we are moving back to using the updateCollections related txHash as the set key
+        else if (activations.isActive(ConsensusRule.RSKIP146) && !activations.isActive(ConsensusRule.RSKIP176)) {
+            // The release transaction may have been created prior to the Consensus Rule activation
+            // therefore it won't have a rskTxHash value, fallback to this transaction's hash
+            rskTxHash = entry.getRskTxHash() == null ? rskTx.getHash() : entry.getRskTxHash();
+        }
+        else {
+            rskTxHash = rskTx.getHash();
+        }
+        return rskTxHash;
+    }
+
+    private void checkIfEntryExistsInTxsWaitingForSignatures(Keccak256 rskTxHash, BtcTransaction btcTransaction, Map<Keccak256, BtcTransaction> txsWaitingForSignatures) {
+        if (txsWaitingForSignatures.containsKey(rskTxHash)) {
+            logger.error(
+                "[checkIfEntryExistsInTxsWaitingForSignatures] An entry for the given rskTxHash {} already exists. Entry " +
+                    "overriding is not allowed for txsWaitingForSignatures map.",
+                rskTxHash
+            );
+            throw new IllegalStateException(
+                String.format(
+                    "An entry for the given rskTxHash {} already exists. Entry overriding is not allowed for txsWaitingForSignatures map.",
```

Comment:
```suggestion
                    "An entry for the given rskTxHash %s already exists. Entry overriding is not allowed for txsWaitingForSignatures map.",
```
The string format works different than the logger. See the code smell you got